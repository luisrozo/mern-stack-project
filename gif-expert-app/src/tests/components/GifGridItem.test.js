import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Tests in <GifGridItem />', () => {

    const title = 'a title';

    const url = 'a url';
    
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    test('it should show the component correctly', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('it should has a paragraph with the title', () => {

        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );

    });

    test('it should have the correct image', () => {

        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });

    test('it should have animate__fadeIn class', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__fadeIn') ).toBe( true );

    })
    

});