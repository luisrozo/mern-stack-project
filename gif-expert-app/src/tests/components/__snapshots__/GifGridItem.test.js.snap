// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> it should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="a title"
    src="a url"
  />
  <p>
     
    a title
     
  </p>
</div>
`;
